x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/JreHttpUrlConnection.java

```diff
@@ -183,6 +186,23 @@ public boolean head() throws ClientException {
         }
     }
 
+    private Function<String, List<String>> getHeaderFields(URLConnection con) {
+        return header -> {
+            // HTTP headers are case-insensitive but the map returned by `URLConnection.getHeaderFields` is case-sensitive.
+            // The linear scan below replicates the linear case-insensitive lookup of `URLConnection.getHeaderField(String)`.
+            List<String> values = new LinkedList<>();
+            int i = 0;
+            String value = con.getHeaderField(i);
+            while (value != null) {
```

Comment:
Just noting that the function would stop iterating on an empty header value? This seems to be legal, though very corner case.

Reply:
I hope that's never the case but the API does not forbid `null` headers. Changing the implementation to use `URLConnection.getHeaderFields()` 👍